Pocket/pocket-ios
# Highlights - improve the highlight on swipe experience

## Summary
* This PR adds some improvements to the highlight on swipe feature:
    - Allows to highlight a paragraph when there are existing partial highlights
    - Allows to delete highlights on swipe (partial or full)
    - Solves a UI issue where images and text could overlap in the Reader, under certain conditions.

## References 
* NA

## Implementation Details
* Add the delete highlights action to `SavedItemViewModel` and conditionally add actions depending one the paragraph state (e.g. if it's fully highlighted, only the delete highlight action will be available)
* Add properties and methods to `ArticleComponentTextCell` concrete implementation and `ArticleComponentTextView` to handle highlight updates

## Test Steps
* Open a saved item and make sure the swipe actions are available and consistent with the content
* Make sure the actions behave as expected

## PR Checklist:
